modules/payment/src/Form/PaymentAddForm.php

```diff
@@ -229,29 +236,30 @@ public static function clearValue(array $element, FormStateInterface $form_state
    *
    * @return array
    *   The built form.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
    */
   protected function buildPaymentMethodForm(array $form, FormStateInterface $form_state) {
-    $payment_method = $this->entityTypeManager
-      ->getStorage('commerce_payment_method')->create([
-        'type' => 'credit_card',
-        'payment_gateway' => $form_state->getValue('payment_gateway'),
-        'uid' => $this->order->getCustomerId(),
-      ]);
-    // Array keys mirror checkout panes to aid in gateway compatibility.
-    $form['contact_information']['email'] = [
-      '#type' => 'hidden',
-      '#value' => $this->order->getEmail(),
-    ];
-    $form['payment_information']['add_payment_method'] = [
+    $payment_method_storage = $this->entityTypeManager->getStorage('commerce_payment_method');
+    $payment_method = $payment_method_storage->create([
+      'type' => $form_state->getValue('payment_method_type'),
+      'payment_gateway' => $form_state->getValue('payment_gateway'),
+      'uid' => $this->order->getCustomerId(),
+    ]);
+    $form['add_payment_method'] = [
       '#type' => 'commerce_payment_gateway_form',
       '#operation' => 'add-payment-method',
       '#default_value' => $payment_method,
+      '#parents' => array_merge($form['#parents'], ['add_payment_method']),
```

Comment:
Which payment gateway are you using to test this? AuthNet has some not-ideal logic that looks for the customer contact information in a structure similar to the checkout panes, hence the contact information array and not using `#parent`. I don't like designing for a particular gateway but until that clears up I don't think it's too particularly harmful to provide that kind of structure here, if the gateway isn't optimal.

Reply:
We don't need to set parents here because the payment gateway form/checkout pane already has #parents. I'm using it like this: https://github.com/steveoliver/commerce_vantiv/compare/admin-order-payments?expand=1. I think this is the only reliable way to target elements.